File: node1t/src/ex1/test04.js
var http = require('http');

// 실습 목표: 응답 내용의 문자셋 지정
// - 요청과 응답에 대해 사용할 수 있는 메서드 
http.createServer(function(req, res) {
    res.writeHead(200, {
    		'Content-Type': 'text/html;charset=UTF-8',
    	});
    res.write('<html><head><title>test04</title></head>\n');
    res.write('<body><h1>test04</h1>\n');
    res.write('<p>안녕하세요!</p>\n');
    res.end('</body></html>\n');
    
}).listen(1337, '127.0.0.1');
console.log('Server running at http://127.0.0.1:1337/');

File: web2t/src/net/bitacademy/java67/step01/Test04.java
package net.bitacademy.java67.step01;

import java.io.IOException;
import java.io.PrintWriter;

import javax.servlet.GenericServlet;
import javax.servlet.ServletException;
import javax.servlet.ServletRequest;
import javax.servlet.ServletResponse;
import javax.servlet.annotation.WebServlet;

// 실습 목표: HTML 출력하기

@WebServlet("/step01/Test04")
public class Test04 extends GenericServlet {
  private static final long serialVersionUID = 1L;
  
  @Override
  public void service(ServletRequest request, ServletResponse response)
      throws ServletException, IOException {
    // Content-type 헤더에 문자집합을 지정하지 않으면, 기본이 ISO-8859-1이다.
    // 이 문자 집합에는 한글이 정의되어 있지 않기 때문에, 제대로된 코드 값으로 바뀔 수가 없다.
    // 코드 값이 없음을 의미하는 '?'로 출력된다.
    response.setContentType("text/html;charset=UTF-8");
    
    PrintWriter out = response.getWriter();
    out.println("<html>");
    out.println("  <head>");
    out.println("    <title>step01/test04</title>");
    out.println("  </head>");
    out.println("  <body>");
    out.println("  <p>하하하 반갑네.</p>");
    out.println("  </body>");
    out.println("</html>");
  }
}



















